ui/src/app/applications/components/application-deployment-history/revision-metadata-rows.tsx

```diff
@@ -1,10 +1,50 @@
 import {DataLoader} from 'argo-ui';
 import * as React from 'react';
-import {Timestamp} from '../../../shared/components/timestamp';
-import {ApplicationSource, RevisionMetadata, ChartDetails} from '../../../shared/models';
+import {Timestamp} from '../../../shared/components';
+import {ApplicationSource, RevisionMetadata, ChartDetails, OCIMetadata} from '../../../shared/models';
 import {services} from '../../../shared/services';
 
 export const RevisionMetadataRows = (props: {applicationName: string; applicationNamespace: string; source: ApplicationSource; index: number; versionId: number | null}) => {
+    if (props?.source?.repoURL?.startsWith('oci://')) {
+        return (
+            <DataLoader
+                input={props}
+                load={input => services.applications.ociMetadata(input.applicationName, input.applicationNamespace, input.source.targetRevision, input.index, input.versionId)}>
+                {(m: OCIMetadata) => (
+                    <div>
+                        <div className='row'>
+                            <div className='columns small-3'>OCI Image:</div>
+                            <div className='columns small-9'>
+                                {props.source.repoURL}&nbsp;
+                                {m.imageUrl && (
+                                    <a
+                                        title={m.imageUrl}
+                                        onClick={e => {
+                                            e.stopPropagation();
+                                            window.open(m.imageUrl);
+                                        }}>
```

Comment:
Need to be super careful about injection here... do we need to do any sanitization on imageUrl?